.github/workflows/run_tests.yml

```diff
@@ -209,8 +222,27 @@ jobs:
         run: |
           uv sync --extra dev --frozen
 
+      - name: Set up Podman
+        if: matrix.docker_engine == 'podman'
+        run: |
+          systemctl start --user podman.socket
+          systemctl status --user podman.socket
+
+          podman --version
+
+          export DOCKER_HOST=unix:///run/user/1001/podman/podman.sock
+          echo "DOCKER_HOST=${DOCKER_HOST}" >> $GITHUB_ENV
+
+      - name: Free some disk space
+        run: |
+          sudo rm -rf /usr/share/dotnet /usr/local/lib/android /opt/ghc \
+                      /usr/local/share/powershell \
+                      /usr/share/swift /usr/local/.ghcup /usr/lib/jvm
```

Comment:
Feels icky to me (but ok if we have no better solution).

Are we sure that cleanup works as expected? Why do we suddenly run out of disk space?

Reply:
> Feels icky to me 

Why?

> Are we sure that cleanup works as expected?

It does. I didn't check how much space it clears, but it is a few GB. This solves the issue of one of the tests erroring with "no space left on device" when unpacking an image.

Reply:
I mean the Docker cleanup we run after each test.

Feels icky because we'll have the same situation again after we add 5 more tests :)

Reply:
> I mean the Docker cleanup we run after each test.

Mostly. I just noticed I have a `unit_vectoradd:latest` after running the E2E tests locally, so maybe there's some cleanup missing?

Reply:
Had another look and the tests pass now, so let's remove this.